### PR TITLE
Cleanup: __init__ files

### PR DESCRIPTION
**Pull request by samuelburbulla**
# Cleanup: __init__ files

### Which issue does this PR tackle?

  - __init__.py files contain implementations.

### How does it solve the problem?

  - Moves all content to submodules.

### How are the changes tested?

  - As before.

## Checklist for Contributors

- [x] Scope: This PR tackles exactly one problem.
- [x] Conventions: The branch follows the `feature/title-slug` convention.
- [x] Conventions: The PR title follows the `Bugfix: Title` convention.
- [x] Coding style: The code passes all pre-commit hooks.
- [x] Documentation: All changes are well-documented.
- [x] Tests: New features are tested and all tests pass successfully.
- [x] Changelog: Updated CHANGELOG.md for new features or breaking changes.
- [x] Review: A suitable reviewer has been assigned.

## Checklist for Reviewers:

- [x] The PR solves the issue it claims to solve and only this one.
- [x] Changes are tested sufficiently and all tests pass.
- [x] Documentation is complete and well-written.
- [x] Changelog has been updated, if necessary.

### CODE REVIEW

File: src/continuity/data/__init__.py [hunk 1]
```diff
@@ -5,85 +5,16 @@
...
-    if use_mps_backend and torch.backends.mps.is_available():
-        device = torch.device("mps")
-
-    if torch.cuda.is_available():
-        device = torch.device("cuda")
-
-    return device
-
-
 device = get_device()
```
Comment (JakobEliasWagner) [inline src/continuity/data/__init__.py:20]: The location of the device inside the `data` package still seems not entirely correct, as parts or the entire operator need to be shipped to the device.

File: src/continuity/data/utility.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+`continuity.data.utility`
+
+Utility functions for data handling.
+"""
+
+import os
+import torch
+
+
+def get_device() -> torch.device:
```
Comment (JakobEliasWagner) [inline src/continuity/data/utility.py:11]: I think this function should belong to the `trainer` package. Not only data but also the operator need to be moved to the device. The `trainer` class takes ownership of that process and should handle the device.

File: src/continuity/operators/common.py [hunk 3]
```diff
@@ -57,6 +63,8 @@ class NeuralNetworkKernel(torch.nn.Module):
     Args:
         kernel_width: Width of kernel network
         kernel_depth: Depth of kernel network
+
+    TODO: As it is implemented, this is a convolution `k(x - y)`, but it should be more general.
```
Comment (JakobEliasWagner) [inline src/continuity/operators/common.py:67]: We should open an issue to keep track of this task.

File: tests/operators/test_deeponet.py [hunk 4]
```diff
@@ -5,7 +5,7 @@
 from continuity.plotting import plot, plot_evaluation
 from torch.utils.data import DataLoader
 from continuity.operators import DeepONet
-from continuity.data import OperatorDataset, Sine
+from continuity.data.sine import OperatorDataset, Sine
```
Comment (JakobEliasWagner) [inline tests/operators/test_deeponet.py:8]: `OperatorDataset` should be imported from `continuity.data` instead of `continuity.data.sine`.

File: src/continuity/operators/neuraloperator.py [hunk 5]
```diff
@@ -1,77 +1,30 @@
...
     Maps continuous functions given as observation to another continuous function and returns point-wise evaluations.
-    The architecture is a stack of continuous convolutions with a lifting layer and a projection layer.
+    The architecture is a stack of continuous kernel integrations with a lifting layer and a projection layer.
 
     *Reference:* N. Kovachki et al. Neural Operator: Learning Maps Between
     Function Spaces With Applications to PDEs. JMLR 24 1-97 (2023)
 
     For now, sensor positions are equal across all layers.
 
+    TODO: This implementation has to be generalized to arbitrary `Operator` layers.
```
Comment (JakobEliasWagner) [inline src/continuity/operators/neuraloperator.py:26]: We should also open an issue for this.

File: tests/operators/test_deeponet.py (hunk 4 above)
Comment (samuelburbulla) [reply tests/operators/test_deeponet.py:9]: Good catch

File: src/continuity/pde/physicsinformed.py [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+            x: Tensor of sensor positions of shape (batch_size, num_sensors, coordinate_dim)
+            u: Tensor of sensor values of shape (batch_size, num_sensors, num_channels)
+            y: Tensor of evaluation coordinates of shape (batch_size, x_size, coordinate_dim)
+            v: Ignored
+        """
+        # Call operator
+        v_pred = op(x, u, y)
+
+        # Get pde loss
+        return self.pde(x, u, y, v_pred)
```
Comment (JakobEliasWagner) [inline src/continuity/pde/physicsinformed.py:76]: the PDE class takes 5 input arguments, but only 4 are shown here. the operator needs to be passed as well or the operator needs to be removed from the PDE. I would prefer the second option.

**Review by JakobEliasWagner — CHANGES_REQUESTED** [review]
the rest looks good to me.

File: src/continuity/pde/physicsinformed.py (hunk 6 above)
Comment (samuelburbulla) [reply src/continuity/pde/physicsinformed.py:85]: I used 4 arguments in the examples for PDEs and also in the physics-informed loss.
For 5 arguments, we have the physics-informed loss.
I fixed the signature of the PDE (which was therefore wrong) to not take the operator as input.
